src/detect-email.c

```diff
@@ -61,6 +62,43 @@ static InspectionBuffer *GetMimeEmailFromData(DetectEngineThreadCtx *det_ctx,
     return buffer;
 }
 
+static int DetectMimeEmailSubjectSetup(DetectEngineCtx *de_ctx, Signature *s, const char *arg)
+{
+    if (DetectBufferSetActiveList(de_ctx, s, g_mime_email_subject_buffer_id) < 0)
+        return -1;
+
+    if (DetectSignatureSetAppProto(s, ALPROTO_SMTP) < 0)
+        return -1;
+
+    return 0;
+}
+
+static InspectionBuffer *GetMimeEmailSubjectData(DetectEngineThreadCtx *det_ctx,
+        const DetectEngineTransforms *transforms, Flow *f, const uint8_t _flow_flags, void *txv,
+        const int list_id)
+{
+    InspectionBuffer *buffer = InspectionBufferGet(det_ctx, list_id);
+    if (buffer->inspect == NULL) {
+        SMTPTransaction *tx = (SMTPTransaction *)txv;
+
+        const uint8_t *b_email_sub = NULL;
+        uint32_t b_email_sub_len = 0;
+
+        if ((tx->mime_state != NULL)) {
+            if (SCDetectMimeEmailGetData(
+                        tx->mime_state, &b_email_sub, &b_email_sub_len, "subject") != 1)
+                return NULL;
+        }
+
+        if (b_email_sub == NULL || b_email_sub_len == 0)
+            return NULL;
```

Comment:
nit @AkakiAlice This looks wrong, we should `return NULL` if `tx->mime_state == NULL)` but not if `SCDetectMimeEmailGetData` returned ok and a 0 length (empty subject is different than absence of subject)